qiskit/ignis/experiments/calibration/methods/basic_1q.py

```diff
@@ -151,3 +155,88 @@ def __init__(self,
                          analysis=analysis,
                          job=job,
                          workflow=data_processing)
+
+
+class SinglePulseSingleParameterGenerator(Generator):
+    """
+    A generator that generates single pulses and scans a single parameter
+    of that pulse. Note that the same pulse may be applied to multiple qubits
+    to perform simultaneous calibration.
+    """
+
+    def __init__(self,
+                 qubits: Union[int, List[int]],
+                 parameters: Dict,
+                 amplitudes: List,
+                 scanned_parameter: str,
+                 pulse: Callable = None):
+        """
+        Args:
+            qubits: List of qubits to which this calibration can be applied.
+            parameters: The arguments for the pulse schedule.
+            amplitudes: A list of amplitudes for which to generate a circuit.
+            scanned_parameter: Name of the scanned parameter. If it is not present
+                in the parameters dictionary as a ParameterExpression it will
+                be added to this dict.
+            pulse: A parametric pulse function used to generate the schedule
+                that is added to the circuits. This defaults to Gaussian and
+                parameters should contain 'duration', and 'sigma'.
+        """
+        super().__init__('Single pulse generator', qubits)
+        self._pulse = pulse
+        self.parameters = parameters
+        self.amplitudes = amplitudes
+        self.scanned_parameter = scanned_parameter
```

Comment:
I think parameter can be decoupled from here. you can generate template (parametrized) circuit in `.circuits` method and bind parameters from `.assign` method (need to implement).

Reply:
I'd like circuits to return a list of circuits that can be executed. Here, parameters should be already assigned.